### DISCUSSION TITLE
Suggest example of branching activities in elsa 3.0

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5588] [Q&A]
To apply a condition before calling a particular activity in an Elsa workflow, you can use the "If" or "Switch" branching activities. These activities allow you to specify conditions and execute different branches of your workflow based on the outcome of those conditions.

Here's how you can use them:

If Activity: Use the "If" activity to specify a condition. If the condition evaluates to true, the activities in the "then" branch will be executed; otherwise, the activities in the "else" branch will be executed.

Switch Activity: Use the "Switch" activity when you have multiple conditions to evaluate. You can define cases based on the value of a given expression, and for each case, you can specify different activities to execute.

To use these activities, you would typically place them in your workflow before the activity you want to conditionally execute. Then, you would configure the condition or cases accordingly, and place the activity you want to execute in the appropriate branch of the "If" or "Switch" activity.

Give example of branching and switch activity

**Comment by glime-ai** [comment]
To apply a condition before calling a particular activity in an Elsa workflow, you can use the "If" or "Switch" branching activities. These activities allow you to specify conditions and execute different branches of your workflow based on the outcome of those conditions.

Here's how you can use them:

**If Activity**: Use the "If" activity to specify a condition. If the condition evaluates to true, the activities in the "then" branch will be executed; otherwise, the activities in the "else" branch will be executed.

Example of using the If activity:
```csharp
var workflow = new Workflow
{
    Root = new If
    {
        Condition = new(true), // Your condition here
        Then = new WriteLine("This will execute if the condition is true"),
        Else = new WriteLine("This will execute if the condition is false")
    }
};
```

**Switch Activity**: Use the "Switch" activity when you have multiple conditions to evaluate. You can define cases based on the value of a given expression, and for each case, you can specify different activities to execute.

Example of using the Switch activity:
```csharp
var workflow = new Workflow
{
    Root = new Switch
    {
        Expression = new Variable<string>("value"), // Your switch expression here
        Cases = {
            { "Case1", new WriteLine("This will execute if the expression is 'Case1'") },
            { "Case2", new WriteLine("This will execute if the expression is 'Case2'") }
        },
        Default = new WriteLine("This will execute if no case matches")
    }
};
```

To use these activities, you would typically place them in your workflow before the activity you want to conditionally execute. Then, you would configure the condition or cases accordingly, and place the activity you want to execute in the appropriate branch of the "If" or "Switch" activity.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
